### PR TITLE
Use delegating containers to reduce duplication

### PR DESCRIPTION
**Pull request by jvandort**
Use Delegating containers to avoid dupliatcating delegating methods. This improves navigation in the IDE, since delegating method implementations only add to noise when looking at different impementations of a given method.

By utilizing delegation, we reduce the number of concrete implementations of methods, thus making it more clear when different container implementations have different behavior from the standard behavior

### CODE REVIEW

**Comment by jvandort** [comment]
@bot-gradle test this

**Comment by jvandort** [comment]
Failing test is unrelated.

**Comment by jvandort** [comment]
@bot-gradle test this

File: subprojects/core/src/main/java/org/gradle/api/internal/DelegatingDomainObjectSet.java [hunk 1]
```diff
@@ -29,10 +29,10 @@
 import java.util.Set;
 
 public class DelegatingDomainObjectSet<T> implements DomainObjectSet<T> {
-    private final DomainObjectSet<T> backingSet;
+    protected final DomainObjectSet<T> delegate;
```
Comment (ghale) [inline subprojects/core/src/main/java/org/gradle/api/internal/DelegatingDomainObjectSet.java:32]: nitpick: Is there a good reason why this is protected and not private with access through a getter?  I didn't see anything but maybe I overlooked it.

**Review by ghale — APPROVED** [review]

File: subprojects/core/src/main/java/org/gradle/api/internal/DelegatingDomainObjectSet.java (hunk 1 above)
Comment (jvandort) [reply subprojects/core/src/main/java/org/gradle/api/internal/DelegatingDomainObjectSet.java:32]: No particular reason. I can change this.

**Comment by jvandort** [comment]
@bot-gradle test this

**Comment by jvandort** [comment]
@bot-gradle test and merge

**Pull request merged**